dataflow_pipeline.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import absolute_import
+
+import argparse
+import logging
+import re
+import timeit
+
+import apache_beam as beam
+from apache_beam.io import ReadFromText
+from apache_beam.io import WriteToText
+from apache_beam.metrics import Metrics
+from apache_beam.metrics.metric import MetricsFilter
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import SetupOptions
+import json
+
+from scripts.google_replicate import exec_google_copy
+
+try:
+    unicode  # pylint: disable=unicode-builtin
+except NameError:
+    unicode = str
+
+FILE_HEADERS = ["id", "filename", "md5", "size", "state", "acl", "project_id"]
+
+# global_config ={"token_path": "./gdc-token.txt",
+#                 "chunk_size_download": 2048000,
+#                 "chunk_size_upload": 20971520}
+
+
+class FileCopyingDoFn(beam.DoFn):
+    def __init__(self, config):
+        super(FileCopyingDoFn, self).__init__()
+        self.global_config = config
+
+    def process(self, element):
+        """Process each line of the manifest file.
+        The element is a line of text.
+        Args:
+          element: the row being processed
+        Returns:
+          The outcome of the copying process. True/False means success/failure
+        """
+        text_line = element.strip()
+        words = text_line.split()
+        fi = dict(zip(FILE_HEADERS, words))
+        fi["size"] = int(fi["size"])
+
+        return [(fi, exec_google_copy(fi, self.global_config))]
+
+
+def format_result(result):
+    (fi, datalog) = result
+    return "%s %s %d %s %s %s %s %s %s" % (
+        fi.get("id"),
+        fi.get("filename"),
+        int(fi.get("size")),
+        fi.get("md5"),
+        fi.get("acl"),
+        fi.get("project_id"),
+        datalog.copy_success,
+        datalog.index_success,
+        datalog.message,
+    )
+
+
+def run(argv=None):
+    """Main entry point; defines and runs the copying pipeline."""
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--input",
+        dest="input",
+        default="./scripts/test_data.txt",
+        help="Input file to process.",
+    )
+    parser.add_argument(
+        "--output",
+        dest="output",
+        required=True,
+        help="Output file to write results to.",
+    )
+    parser.add_argument(
+        "--global_config", dest="global_config", help="global configuration"
```

Comment:
what format should this global configuration be in? What are you passing? A file name? can you add more details to the `help` section explaining these things?

Reply:
+1 to this